js/src/utils/toScopeState.test.js

```diff
@@ -0,0 +1,156 @@
+	// Manage AdWords campaigns
+	AD_WORDS: 'https://www.googleapis.com/auth/adwords',
+};
+
+describe( 'toScopeState', () => {
+	let contentScopes;
+	let siteVerificationScopes;
+	let gmcScopes;
+	let adsScopes;
+	let allScopes;
+
+	beforeEach( () => {
+		contentScopes = [ SCOPE.CONTENT ];
+		siteVerificationScopes = [ SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
+		gmcScopes = [ SCOPE.CONTENT, SCOPE.SITE_VERIFICATION_VERIFY_ONLY ];
+		adsScopes = [ SCOPE.AD_WORDS ];
+		allScopes = [
+			SCOPE.CONTENT,
+			SCOPE.SITE_VERIFICATION_VERIFY_ONLY,
+			SCOPE.AD_WORDS,
+		];
+	} );
+
+	it( 'should return an object structure containing three properties', () => {
+		const scopeState = toScopeState( false, [] );
+
+		expect( scopeState ).toHaveProperty( 'gmcRequired' );
+		expect( scopeState ).toHaveProperty( 'adsRequired' );
+		expect( scopeState ).toHaveProperty( 'glaRequired' );
+	} );
+
+	it( 'when the `scopes` parameter is not given, should get `false` for all results', () => {
+		expect( toScopeState( false ) ).toMatchObject( {
+			gmcRequired: false,
+			adsRequired: false,
+			glaRequired: false,
+		} );
+		expect( toScopeState( true ) ).toMatchObject( {
+			gmcRequired: false,
+			adsRequired: false,
+			glaRequired: false,
+		} );
+	} );
+
+	describe.each( [ [ false ], [ true ] ] )(
+		'For `gmcRequired`, if the parameter `adsSetupComplete` = %p',
+		( adsSetupComplete ) => {
+			it( 'and the `scopes` contains GMC required scopes, should be `true`', () => {
+				expect(
+					toScopeState( adsSetupComplete, gmcScopes ).gmcRequired
+				).toBe( true );
+			} );
+
+			it( "and the `scopes` doesn't contains GMC required scopes, should be `false`", () => {
+				expect( toScopeState( adsSetupComplete, [] ).gmcRequired ).toBe(
+					false
+				);
+				expect(
+					toScopeState( adsSetupComplete, contentScopes ).gmcRequired
+				).toBe( false );
+				expect(
+					toScopeState( adsSetupComplete, siteVerificationScopes )
+						.gmcRequired
+				).toBe( false );
+				expect(
+					toScopeState( adsSetupComplete, adsScopes ).gmcRequired
+				).toBe( false );
+			} );
+		}
+	);
+
+	describe.each( [ [ false ], [ true ] ] )(
+		'For `adsRequired`, if the parameter `adsSetupComplete` = %p',
+		( adsSetupComplete ) => {
+			it( 'and the `scopes` contains Google Ads required scopes, should be `true`', () => {
+				expect(
+					toScopeState( adsSetupComplete, adsScopes ).adsRequired
+				).toBe( true );
+			} );
+
+			it( "and the `scopes` doesn't contains Google Ads required scopes, should be `false`", () => {
+				expect( toScopeState( adsSetupComplete, [] ).adsRequired ).toBe(
+					false
+				);
+				expect(
+					toScopeState( adsSetupComplete, contentScopes ).adsRequired
+				).toBe( false );
+				expect(
+					toScopeState( adsSetupComplete, siteVerificationScopes )
+						.adsRequired
+				).toBe( false );
+				expect(
+					toScopeState( adsSetupComplete, gmcScopes ).adsRequired
+				).toBe( false );
+			} );
+		}
+	);
```

Comment:
❔ By looking at these two `describe.each` blocks here, we would notice that for `gmcRequired` and `adsRequired`, the `toScopeState`'s first parameter `adsSetupComplete` - whether it is `true` or `false` - does not matter. 

I feel like the API isn't really that nice, and there should be a better way. 🤔 

I looked into the application code. We don't have any usage of `adsRequired`. We have one usage of `gmcRequired` which is used in the following...

https://github.com/woocommerce/google-listings-and-ads/blob/0cad7cd8defcfbd433f91ea963b12820cba41f24/js/src/setup-mc/setup-stepper/setup-accounts/index.js#L40-L42

... which I think can be replaced with `glaRequired` and it would work fine, which means that `gmcRequired` is not needed too. 🤔 

If `gmcRequired` and `adsRequired` are not needed, then we can simplify `toScopeState` (and its tests), e.g. instead of `toScopeState (arg1, arg2)` which returns an object, it can be `hasRequiredPermissions (arg1, arg2)` which returns a boolean value (i.e. `glaRequired`).

Does this make sense?

Reply:
`adsRequired` is used [here](https://github.com/woocommerce/google-listings-and-ads/commit/523fb30191e63c64d772c39e8a0efc49e607854a#diff-71606b9acaa50832ca8d6c49bb920d55cd2a1c2e02f13ac9028f4f468b0c5d9eR19) (PR #1069)

Reply:
> `adsRequired` is used [here](https://github.com/woocommerce/google-listings-and-ads/commit/523fb30191e63c64d772c39e8a0efc49e607854a#diff-71606b9acaa50832ca8d6c49bb920d55cd2a1c2e02f13ac9028f4f468b0c5d9eR19) (PR #1069)

That `adsRequired` can be replaced with `glaRequired` and things would still work the same too, right? 🤔

Reply:
Replace `adsRequired ` with `glaRequired` could get the correct results.

However, the semantics are not consistent with the purpose, because on the Google Ads setup page, the only permission that needs to be checked is the ad permission.

And even if the replacement works correctly at this moment, it'd hide a potential problem that if new features are added that require new permissions and also added into `glaRequired`, then the logic [here](https://github.com/woocommerce/google-listings-and-ads/commit/523fb30191e63c64d772c39e8a0efc49e607854a#diff-71606b9acaa50832ca8d6c49bb920d55cd2a1c2e02f13ac9028f4f468b0c5d9eR19) might become incorrect. I lean toward keeping it as is.